eventuate-tram-quarkus-in-memory/src/test/java/io/eventuate/tram/quarkus/inmemory/InMemoryMessageProducerTest.java

```diff
@@ -3,112 +3,115 @@
+public class InMemoryMessageProducerTest extends AbstractInMemoryMessageProducerTest {
   @Inject
-  InMemoryMessageProducer inMemoryMessageProducer;
+  MessageProducer messageProducer;
 
   @Inject
-  InMemoryMessageConsumer inMemoryMessageConsumer;
+  MessageConsumer messageConsumer;
 
   @Inject
-  EventuateTransactionTemplate eventuateTransactionTemplate;
+  TransactionManager transactionManager;
 
+  @Override
+  protected MessageProducer getMessageProducer() {
+    return messageProducer;
+  }
 
-  class MyMessageHandler implements MessageHandler {
+  @Override
+  protected MessageConsumer getMessageConsumer() {
+    return messageConsumer;
+  }
 
-    private BlockingQueue<String> queue = new LinkedBlockingDeque<>();
+  @Override
+  protected void executeInTransaction(Consumer<Runnable> consumer) {
+    try {
+      transactionManager.begin();
 
-    @Override
-    public void accept(Message message) {
-      try {
-        queue.put(message.getPayload());
-      } catch (InterruptedException e) {
-        throw new RuntimeException(e);
-      }
-    }
+      AtomicBoolean rollback = new AtomicBoolean(false);
+
+      consumer.accept(() -> {
+          rollback.set(true);
+      });
 
-    void shouldReceiveMessage(String payload) {
-      String m;
-      try {
-        while ((m = queue.poll(1, TimeUnit.SECONDS)) != null) {
-          if (payload.equals(m))
-            return;
-        }
-        fail("Didn't find message with payload: " + payload);
-      } catch (InterruptedException e) {
-        throw new RuntimeException(e);
+      if (rollback.get()) {
+        transactionManager.rollback();
+      } else {
+        transactionManager.commit();
       }
+
+    } catch (NotSupportedException | SystemException | RollbackException | HeuristicMixedException | HeuristicRollbackException e) {
+      throw new RuntimeException(e);
     }
   }
 
+  @Override
   @BeforeEach
   public void setUp() {
-    subscriberId = "subscriberId-" + System.currentTimeMillis();
-    destination = "destination-" + System.currentTimeMillis();
-    payload = "payload-" + System.currentTimeMillis();
-    mh = new MyMessageHandler();
+    super.setUp();
   }
 
+  @Override
   @Test
   public void shouldDeliverToMatchingSubscribers() {
-
-    inMemoryMessageConsumer.subscribe(subscriberId, Collections.singleton(destination), mh);
-
-    Message m = makeMessage();
-    inMemoryMessageProducer.send(m);
-    assertNotNull(m.getId());
-    mh.shouldReceiveMessage(payload);
-
+    super.shouldDeliverToMatchingSubscribers();
   }
 
+  @Override
   @Test
   public void shouldSetIdWithinTransaction() {
-    Message m = makeMessage();
-    eventuateTransactionTemplate.executeInTransaction(() -> {
-      inMemoryMessageProducer.send(m);
-      assertNotNull(m.getId());
-      return null;
-    });
+    super.shouldSetIdWithinTransaction();
   }
 
+  @Override
   @Test
   public void shouldDeliverToWildcardSubscribers() {
+    super.shouldDeliverToWildcardSubscribers();
```

Comment:
I'm confused. All three concrete subclasses of `AbstractInMemoryMessageProducerTest` have test methods that look like this: one-liners that call super. Can't these methods be removed and the superclass methods be annotated with @Test?

Reply:
@cer spring and quarkus/micronaut uses different test api. org.junit.jupiter.api.Test and org.junit.Test correspondingly.